PyTDF/Node.py

```diff
@@ -55,20 +54,32 @@ def _dfs(self, node, prev_object= None):
             self._dfs(n, node.value)
 
     
-    def _graph_prune_util(self, node):
+    def _graph_prune(self, node=None):
```

Comment:
Why do you need the node parameter? Can't you just prune `self`? In the first call, `self` will be the head node, later in the recursion it will be one of the children. This way you do not need the `if not node` code.

Reply:
This is a very good point. Thanks ! 👍